test/e2e/testing-manifests/sample-device-plugin.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: sample-device-plugin-beta
+  namespace: kube-system
+  labels:
+    k8s-app: sample-device-plugin
+spec:
+  selector:
+    matchLabels:
+      k8s-app: sample-device-plugin
+  template:
+    metadata:
+      labels:
+        k8s-app: sample-device-plugin
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+    spec:
+      priorityClassName: system-node-critical
+      tolerations:
+      - operator: "Exists"
+        effect: "NoExecute"
+      - operator: "Exists"
+        effect: "NoSchedule"
+      volumes:
+      - name: device-plugin
+        hostPath:
+          path: /var/lib/kubelet/device-plugins
+      - name: plugins-registry-probe-mode
+        hostPath:
+          path: /var/lib/kubelet/plugins_registry 
+      - name: dev
+        hostPath:
+          path: /dev
+      containers:
+      - image: "docker.io/vikaschoudhary16/sample-device-plugin-amd64:1.0" # TODO(vikasc): Use image from a k8s-owned repo 
```

Comment:
Lets build the image from `test/images/sample-device-plugins` and use that here!

/cc @ixdy

Reply:
@mkumatag build is created from `test/images/sample-device-plugins` only. As mentioned in comments, after this PR gets merged, image will be pushed to official registry and link would be updated here.

Reply:
@vikaschoudhary16 usually, the image will be built and pushed first before any reference. We can even have a separate PR for just image and get it populated before the reference. @ixdy what do you think?

One more thing: please refer the multi-arch fat manifest instead of any `-arch` specific image.

Reply:
@mkumatag 
>please refer the multi-arch fat manifest instead of any -arch specific image.

No idea what you mean by ^ . What is a "multi-arch fat manifest"? Can you share reference link please?

Reply:
> Regarding image push, @dashpole suggested that we could push it post this PR merge.
> [#67928 (comment)](https://github.com/kubernetes/kubernetes/pull/67928#issuecomment-451611996)

Most of the folks first submit the code for the image, push images and then refer that here in the tests. That is the process been followed so far.

I'll leave the final decisions to @ixdy @luxas.

Reply:
@mkumatag See [here](https://github.com/kubernetes/kubernetes/pull/67928/files#diff-92a015323f660c3ccafaf84bce3701a6R2). It is referring multiple platforms. If this is not what you meant, can you please share link to any such Dockerfile that is already being used in k8s?

Reply:
@mkumatag is suggesting that this be `gcr.io/kubernetes-e2e-test-images/sample-device-plugin:1.0` with no -ARCH suffix.

I also agree that it might be worth separating the image and e2e test changes into two separate PRs.

Reply:
@ixdy PTAL at seperate PR for image changes, https://github.com/kubernetes/kubernetes/pull/73511